### PR TITLE
[Incito Reader, Event Tracking] Implement "Incito Publication Section Opened" event

### LINKED ISSUE #164: [Incito Reader, Event Tracking] Implement "Incito Publication Section Opened" event
Spec is here: https://github.com/tjek/TjekTeams/issues/697

### PR DESCRIPTION
**Pull request by klarstrup**
Closes #164

### CODE REVIEW

File: lib/incito-browser/incito.ts [hunk 1]
```diff
@@ -648,7 +648,26 @@ function renderView(view, canLazyload) {
...
 export default class Incito extends MicroEvent {
+    declare bind: <N extends keyof IncitoEventMap>(
+        name: N,
+        cb: (...args: IncitoEventMap[N]) => void
+    ) => void;
+    declare unbind: <N extends keyof IncitoEventMap>(
+        name: N,
+        cb: (...args: IncitoEventMap[N]) => any
+    ) => void;
+    declare trigger: <N extends keyof IncitoEventMap>(
```
Comment (mortenbo) [inline lib/incito-browser/incito.ts]: I feel like this looks impossible to read/maintain 😅 Can we clean this up/simplify?

Comment (klarstrup) [reply lib/incito-browser/incito.ts]: Yeah and I want to do that, but it's out of scope of this PR so I restrained myself to specifically typing the stuff I'm adding

**Review by mortenbo — APPROVED** [review]
Cool - this looks good. We haven't added the event to Wolf yet but we'll do that tomorrow. Will ping you.

File: lib/incito-browser/incito.ts (hunk 1 above)
Comment (mortenbo) [reply lib/incito-browser/incito.ts]: Roger 👍

File: lib/incito-browser/incito.ts [hunk 2]
```diff
@@ -648,7 +648,12 @@ function renderView(view, canLazyload) {
     return {tagName, contents, attrs};
 }
 
-export default class Incito extends MicroEvent<{started: []; destroyed: []}> {
+export default class Incito extends MicroEvent<{
+    started: [];
+    destroyed: [];
+    sectionVisible: [{sectionId: string; sectionPosition: number}];
+    sectionHidden: [{sectionId: string; sectionPosition: number}];
+}> {
```
Comment (klarstrup) [inline lib/incito-browser/incito.ts:656]: @mortenbo how's this for size

Comment (mortenbo) [reply lib/incito-browser/incito.ts:656]: @klarstrup Looks very naice 🤩

**Review by mateuszjurewicz — APPROVED** [review]
The included fields look good, the 500ms lower limit is in place, I think it's ok as far as I can tell :)

**Pull request merged**